tests/operatordict_test.py

```diff
@@ -0,0 +1,29 @@
+import unittest
+
+from clvm.operators import OperatorDict
+
+
+class OperatorDictTest(unittest.TestCase):
+    def test_operatordict_constructor(self):
+        """Constructing should fail if quote or apply are not specified,
+           either by object property or by keyword argument.
+           Note that they cannot be specified in the operator dictionary itself.
+        """
+        d = {1: "hello", 2: "goodbye"}
+        with self.assertRaises(AttributeError):
+            o = OperatorDict(d)
+        with self.assertRaises(AttributeError):
+            o = OperatorDict(d, apply=1)
+        with self.assertRaises(AttributeError):
+            o = OperatorDict(d, quote=1)
+        o = OperatorDict(d, apply=1, quote=2)
+        print(o)
+        # Why does the constructed Operator dict contain entries for "apply":1 and "quote":2 ?
+        # assert d == o
+        assert o.apply_atom == 1
+        assert o.quote_atom == 2
```

Comment:
how does this differ from `self.assertEqual()`? is the non-member assert to be preferred?

Reply:
Yeah, I'd recommend using `self.assertEqual` although I'm not exactly sure why.

Reply:
self.assertEqual is a little better - display output on failure is different